index.js

```diff
@@ -56,35 +60,32 @@ export default class PCancelable {
 					get: () => this._rejectOnCancel,
 					set: boolean => {
 						this._rejectOnCancel = boolean;
-					}
-				}
+					},
+				},
 			});
 
 			executor(onResolve, onReject, onCancel);
 		});
 	}
 
 	then(onFulfilled, onRejected) {
-		// eslint-disable-next-line promise/prefer-await-to-then
 		return this._promise.then(onFulfilled, onRejected);
 	}
 
 	catch(onRejected) {
-		// eslint-disable-next-line promise/prefer-await-to-then
 		return this._promise.catch(onRejected);
 	}
 
 	finally(onFinally) {
-		// eslint-disable-next-line promise/prefer-await-to-then
 		return this._promise.finally(onFinally);
 	}
 
 	cancel(reason) {
-		if (!this._isPending || this._isCanceled) {
+		if (this._state !== promiseState.pending || this._state === promiseState.canceled) {
```

Comment:
The first check is moot now.

Reply:
I think probably the second check is moot here since `cancel` should do nothing when the promise is `resolved`, `rejected` or `canceled` already.

Reply:
Ah, yes.